rfcs/rfc-manage-staff-role-and-account.md

```diff
@@ -0,0 +1,75 @@
+
+This is an RFC draft to discuss more suitable approaches 
+
+# Basic example
+
+If the proposal involves a new or changed API, component, etc., include a basic code example.
+Omit this section if it's not applicable.
+
+# Motivation
+SBC staff want to access the staff dashboard to search for SP/GP filing for clients, They may have to register SP/GP companies on behalf of the client
+
+# Solution Options
+## 1) Use a new role for sbc staff from keycloak
+
+### Summary
+Keycloak Roles are currently used across the application to determine who can access the staff dashboard.
+We can create a new keycloak role for sbc-staff and in code, we can use this group to give permission for required components/pages.
+In this way, sbc-staff can do alterations for the organization on behalf of the user, and sbc-staff can use their GOVM account to create new SP/GP for Users
+We can create a new GOVM account and add all BC Registry staff into that GOVM account where our staff can access the products as sbc-staffs do.
+
+### Detailed design
+1. Have to create a new role in keycloak for sbc-staff and
+2. In code we have to check at all places where we need to give permission for the new role, like staffed
+3. In addition to account (where they are part), we can show the staff dashboard in the menu, where they can switch between staff dashboard and account 
+4. When sbc-staff/Registry staff access via staff dashboard, they can access all the accounts in our system (behalf of the user)
+5.  sbc-staff/Registry staff can use their  GOVM account to create new organization/filing etc .. on behalf of clients and they can manage it.
+6. sbc-staff/Registry staff  can use their  GOVM account  to access products like PPR/Name requests to help the clients
+7. As in existing we can have multiple staff groups to limit permission (search only/view/edit permissions)
+
+
+This approach will have a high dev impact on sbc-auth and pay-api, less dev impact on and all partner applications (can directly check user role from keycloak tokens)
+
+
+## 2) Bring all Registry staff into one account and connect accounts using account delegation
+
+### Summary
+Currently, the Registry staff is not part of any accounts, and the Registry staff's role is being verified directly from keycloak. As a part of this solution, all users will be under at least one account and Registry staff will be deligated to all accounts according to access permissions. Then we will handle granular permission for staff from sbc-auth like end-users.
+The same will apply to sbc-staff also. Sbc-staff will delegate access to accounts which they want to work on behalf of the client
+
+It will be similar to the below RFC (instead of entity, the account will be deligated to another account)
+https://github.com/bcgov/entity/blob/master/rfcs/rfc-delegation-and-permissions.md
+
+### Detailed design
+1. Account will be deligated to another account
+2. All permissions will be granular level, Auth API will control Permissions
```

Comment:
The current design is auth is supposed to provide _labels_ (aka roles) that are then roles in the JWT that downstream apps can get and interpret. Auth provides some endpoints to do things like, is this label assigned to this user.
We try to avoid granular within the connected apps, so that other agencies can use this in a scalable manner.

Reply:
yeah, agreed.  I don't think we want to have to call auth to determine what a user is allowed to do for every little thing.  The connected apps can decide how to limit functionality based off of the role extracted from the JWT.

Reply:
We need to use granular level permission only if we are using account delegation. (which can be grouped and used for checking similar to groups in keycloak), Currently, we are using one endpoint in sbc-auth to get account level permission for users (customers to know what type of access they have like team-member/admin/co-ordinator). For staff, we are using only keycloak group, for a different level of permission. 
when we use account delegation, I am not sure we can use this, since permission will be related to the user/account, not only to the user (I don't know if we can give keycloack user/account multiple permission. please suggest).

Note: in sbc-auth we have user/account level permission, If you are a team member only (not admin/co-coordinator) you can't change payment options ie: user have limited permission only in that account